core/store/src/config.rs

```diff
@@ -157,22 +155,19 @@ impl<'a> StoreOpener<'a> {
 
     /// Returns version of the database; or `None` if it does not exist.
     pub fn get_version_if_exists(&self) -> Result<Option<DbVersion>, crate::db::DBError> {
-        std::fs::canonicalize(&self.path)
-            .ok()
-            .map(|path| crate::RocksDB::get_version(&path))
-            .transpose()
+        self.check_if_exists().then(|| crate::RocksDB::get_version(&self.path)).transpose()
```

Comment:
Maybe just `if !self.check_if_exists() { return Ok(None) }` so that the reader doesn't have to solve combinator puzzle?